test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -506,7 +506,7 @@ protected bool CheckEqualityFromPathsCore(string relPath, string basePath, strin
 
         private void GetNumbersFromFile(ref string firstString, ref string secondString, int digitsOfPrecision)
         {
-            Regex _matchNumer = new Regex(@"\b[0-9]+\.?[0-9]*(E-[0-9]*)?\b", RegexOptions.IgnoreCase | RegexOptions.Compiled);
+            Regex _matchNumer = new Regex(@"-?[0-9]+\.?[0-9]*(E-[0-9]*)?\b", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```

Comment:
Do we still want the `\b` at the beginning?

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-escapes-in-regular-expressions

\b | In a [character_group] character class, matches a backspace, \u0008. (See Character Classes.) Outside a character class, \b is an anchor that matches a word boundary. (See Anchors.)
-- | --


How about the string: `abc123`. Doesn't the above regex now match `123`, when before it didn't?

Reply:
There are 2 cases that need to be discussed

1.  abc123 - earlier there was no match with regex and we were doing an exact match. Now 123 will be picked up by the regex and we will be doing a floating point match. In this case floating point match will behave like an exact match due to no decimal point.

2. netcoreapp2.1 - This case might be a problem. Earlier we were matching just 1 and doing a floating point match which behaved like an exact match but now we are matching 2.1 and depending upon number of digits of precision it might match something else. eg netcoreapp2.1321312 and netcoreapp2.1321012

However they wont be any mathematical result numbers which is the purpose of this matching. we are just relaxing the restriction on this case. which I think is okay in this case.

Reply:
> we are just relaxing the restriction on this case. which I think is okay in this case.

I'm not sure I agree. Why are we relaxing the restriction? We can add support for `-` without taking away the `\b`. We should keep the current behavior (only matching numbers) unless we have an explicit reason for doing it.